_pages/documentation/gem5-stdlib/7-using-local-data-sources.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: documentation
+title: Using local Data sources in gem5
+parent: gem5-standard-library
+doc: gem5 documentation
+permalink: /documentation/gem5-stdlib/using-local-resources
+author: Harshil Patel
+---
+
+gem5 supports using local data sources in the form of a MongoDB Atlas and JSON datasource. To utilize data sources other than the main gem5 resources database, you will need to override the gem5-resources-config.
+
+MongoDB Config Format:
+```json
+{
+    "sources":{
+        "example-atlas-config": {
+            "dataSource": "datasource name",
+            "database": "database name",
+            "collection": "collection name",
+            "url": "Atlas data API URL",
+            "authUrl": "Atlas authentication URL",
+            "apiKey": "API key",
+            "isMongo": true
+        }       
+    }
+}
+```
+JSON Config Format:
+
+```json
+{
+    "sources":{
+        "example-json-config": {
+            "url": "local path to JSON file or URL to a JSON file",
+            "isMongo": false
+        }
+    }
+}
+```
+### Updating gem5 Resources Config:
+
+You can update the gem5 resources config in the following ways:
+
+Set the GEM5_CONFIG environment variable to point to a new config file. This will override the default config.
+
+Have a file named gem5-config.json in the current working directory. This will also override the default config.
+
+Use the GEM5_RESOURCE_JSON environment variable to override the default config to use a JSON file, or use the GEM5_RESOURCE_JSON_APPEND environment variable to add a JSON file to the current config. If the config is overridden by the above two methods, then these environment variables will override or modify that config.
+
+### Using Multiple Data Sources
+
+gem5 supports the use of more than one data source. The config structure is as follows:
```

Comment:
We should use "resource configuration" to refer to the json file containing the resource metadata. Saying "configuration" alone is confusing because we also use "configuration" in many other places, like "python system configuration" for the python file inputted to gem5.

```suggestion
gem5 supports the use of more than one data source. The structure of resource configuration is as follows:
```